src/domains/vote/components/DelegateTable/DelegateRow/DelegateRow.tsx

```diff
@@ -207,9 +211,14 @@ export const DelegateRow = ({
 				isCompact={isCompact}
 			>
 				<Avatar size={isCompact ? "xs" : "lg"} className="-ml-0.5" address={delegate.address()} noShadow />
-				<div className="relative grow">
-					<span className="absolute flex w-full items-center">
-						<div className="overflow-hidden text-ellipsis">{delegate.username()}</div>
+				<div className="relative grow ">
+					<span className="absolute flex w-full items-center whitespace-nowrap" ref={ref}>
+						{delegate.username() ? (
+							<div className="overflow-hidden text-ellipsis">{delegate.username()}</div>
+						): (
+							<TruncateMiddleDynamic value={delegate.address()} availableWidth={width} />
+						)} 
```

Comment:
will leave this for now, but later on we need a generic handler as it will apply to regular wallets and other parts of the UI as well